File: hella/src/main/java/com/lydiaschiff/hella/renderer/DefaultRsRenderer.java
package com.lydiaschiff.hella.renderer;

import android.renderscript.Allocation;
import android.renderscript.RenderScript;
import androidx.annotation.RequiresApi;

import com.lydiaschiff.hella.RsRenderer;

/**
 * Render an image allocation without modification.
 */
@RequiresApi(18)
public class DefaultRsRenderer implements RsRenderer {

    @Override
    public void renderFrame(RenderScript rs, Allocation in, Allocation out) {
        out.copyFrom(in);
    }

    @Override
    public String getName() {
        return "default (no edits applied)";
    }

    @Override
    public boolean canRenderInPlace() {
        return true;
    }
}

File: hella/src/main/java/com/lydiaschiff/hella/renderer/DefaultRsRenderer.kt
package com.lydiaschiff.hella.renderer

import android.renderscript.Allocation
import android.renderscript.RenderScript
import androidx.annotation.RequiresApi
import com.lydiaschiff.hella.RsRenderer

/**
 * Render an image allocation without modification.
 */
@RequiresApi(18)
class DefaultRsRenderer : RsRenderer {

    override fun renderFrame(rs: RenderScript, inAlloc: Allocation, outAlloc: Allocation) {
        outAlloc.copyFrom(inAlloc)
    }

    override val name = "default (no edits applied)"

    override fun canRenderInPlace() = true
}